src/server/generic_family.cc

```diff
@@ -597,6 +597,10 @@ void OpScan(const OpArgs& op_args, const ScanOpts& scan_opts, uint64_t* cursor,
   auto& db_slice = op_args.GetDbSlice();
   DCHECK(db_slice.IsDbValid(op_args.db_cntx.db_index));
 
+  // We need to make sure we don't preempt below, because we don't hold any locks to the keys
```

Comment:
I dont think its related to holding locks for keys
We do not allow preemption in traverse callback here, because other fiber can edit the bucket at the time we yield
Please write:
Because ScanCb can preempt due to journaling expired entries we need to make sure that we enter the callback in a timing when journaling will not cause preemtion

Reply:
It's both. Let's ignore bucket splits when we preempt. Now let's say the bucket has 5 slots. 
We iterate the first and second and on the third we call `ExpireIfNeeded` from `ScanCb`. The issue is that we preempt `when we write to the journal` but we have `not` yet deleted the `entry` from the dash table. If we preempt on the first step, then we could get a transaction on the same key that just does a blind update. What's the problem there? That once we resume we will expire the `update`. That's a bug and its a side effect of not holding the locks to the relevant key.

Why for example it is ok to call `ExpireIfNeeded` and preempt in `Find` ? Well, because this is transactional and we hold the locks so even if we preempt we don't risk the corner case described above.

I will leave a big comment here.

Reply:
hmm I think I might be wrong because if we `preempt` when we write the journal change, the transaction that will try to modify the same key will block on `PreUpdate()` since it will try to call the registered callback that locks the same mutex and will block. In otherwords, the write to journal within ScanCb will `syncronize with PreUpdate()`